source/main/utils/preload.js

```diff
@@ -0,0 +1,21 @@
+const _process = process;
+
+process.once('loaded', () => {
+  global.process = {
+    env: {
+      NODE_ENV: _process.env.NODE_ENV,
+      NETWORK: _process.env.NETWORK || 'development'
+    },
+    umask: _process.umask,
+    version: _process.version,
+    cwd: _process.cwd,
+    platform: _process.platform
+  };
+  global.Buffer = Buffer;
+  global.require = require;
+  // ESLint will warn about any use of eval(), even this one
+  // eslint-disable-next-line
+  global.eval = () => {
+    throw new Error('This app does not support window.eval().');
+  };
+});
```

Comment:
I really like the preload script idea … why not copy the whole environment here and pick it up in `environment.js` if it exists? This way we could remove the IPC channel and weird sprinkle of environment logic all over the stores (which binds the stores to IPC even more)